app/src/androidTest/java/ch/epfl/sdp/drone3d/service/drone/DroneFactoryTest.kt

```diff
@@ -20,15 +21,27 @@ class DroneFactoryTest {
     @get:Rule
     val instantTaskExecutorRule = InstantTaskExecutorRule()
 
+    //This test passes and I would like to know your thoughts on it
     @Test
-    fun createSimulationWorks() {
+    fun createSimulationWithWronglyParsedDataSucceeds() {
         val factory = DroneServerFactoryImpl()
-        val instance = factory.createSimulation("ip", "port")
+        val instance = factory.createSimulation("sim", "ip")
 
-        assertThat(instance.server, notNullValue())
-        assertThat(instance.instance, notNullValue())
+        assertThat(instance?.server, notNullValue())
+        assertThat(instance?.instance, notNullValue())
 
-        instance.server.stop()
+        instance?.server?.stop()
+    }
```

Comment:
I think the problem comes from the mavsdk library, and we have a check earlier to assure ourselves that the values are valid

Reply:
The comment is about the removed test

Reply:
I removed the test